packages/compiler/src/extractor/extract.ts

```diff
@@ -77,6 +82,7 @@ export const extractProps = (
   // However, we do not want to apply the Box theme in those cases.
   if (componentName === "Box" && isDefault) {
     for (const prop in componentVariantProps) {
+      // eslint-disable-next-line no-prototype-builtins -- FIXME
       if (componentVariantProps.hasOwnProperty(prop)) {
```

Comment:
Note: this shows`eslint:recommended` (specifically [this rule](https://eslint.org/docs/latest/rules/no-prototype-builtins)) can be useful to have 🎉  (well this is just an unlikely edge case but  `componentVariantProps` might. be `{hasOwnProperty: "something"}`, still better to avoid this pattern)

If we can only support above Node.js v16 recent minor, we can just use [`Object.hasOwn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn)

If we care old Node.js we can use `Object.prototype.hasOwnProperty.call(componentVariantProps, prop);`


(I can create follow up PR to fix some of FIXME comments added in this PR. It's better to make sure that this PR does not change any runtime behavior)

Reply:
Follow up MR https://github.com/poteboy/kuma-ui/pull/244